### ISSUE TITLE
Compare scripts should be build using language defaults

### ISSUE THREAD

**Issue by RagnarGrootKoerkamp** [DOMjudge/domjudge#1238]
Currently the judgedaemon has [hardcoded compile commands](https://github.com/DOMjudge/domjudge/blob/ea466079fc6be5aed904fdee741a52ec3ea05085/judge/judgedaemon.main.php#L328-L347) for executables (output validators / interactors).

It would be nicer if this reused the configured language compile scripts from the DOMserver settings, i.e. the default [`build`](https://github.com/DOMjudge/domjudge/blob/main/sql/files/defaultdata/cpp/build) and [`run`](https://github.com/DOMjudge/domjudge/blob/main/sql/files/defaultdata/cpp/run) script for the language.

**Comment by RagnarGrootKoerkamp** [comment]
I think I should be able to reuse most of the existing [`compile` function](https://github.com/DOMjudge/domjudge/blob/main/judge/judgedaemon.main.php#L1007) for team submissions.

The tricky point is mapping an executable to a compile script/id. It seems that currently no language detection is done on the judgedaemons, so they would need to fetch this information from the database first, right? (Of course we could still keep the [switch](https://github.com/DOMjudge/domjudge/blob/main/judge/judgedaemon.main.php#L351) over extensions, but that feels wrong.)

Then, we need the  up date `compile_config` for all languages so we can know the most recent [hash](https://github.com/DOMjudge/domjudge/blob/main/judge/judgedaemon.main.php#L1075) and fetch the executable.

Basically, it seems that we need to query the database each time before compiling an executable. Is that possible, and does it sounds like the right approach?


----

Also, here's a pedantic case: Suppose the executable for the `xyz` language only consist of a `main.xyz` file that should be automatically compiled to a `run` binary used to compile files in the `xyz` language. Now to do this, we need to fetch the compile script for the `xyz` language, but that first needs to be build, and so on.

**Comment by RagnarGrootKoerkamp** [comment]
Oh, or we could do language detection for executables early on and store it in the DB, and then send this language, together with compile config when the executable is fetched [here](https://github.com/DOMjudge/domjudge/blob/main/judge/judgedaemon.main.php#L295)
